conf/modules/aligner.config

```diff
@@ -44,8 +44,41 @@ process {
                 mode: params.publish_dir_mode,
                 path: { "${params.outdir}/preprocessing/" },
                 pattern: "*bam",
-                // Only save if (save_mapped OR (no_markduplicates AND save_output_as_bam)) AND only a single BAM file per sample
-                saveAs: { (params.save_output_as_bam && (params.save_mapped || params.skip_tools && params.skip_tools.split(',').contains('markduplicates'))) && (meta.size * meta.num_lanes == 1) ? "mapped/${meta.id}/${it}" : null }
+                // Only save if save_output_as_bam AND
+                // (save_mapped OR no_markduplicates OR sentieon_dedup) AND
+                // only a single BAM file per sample
+                saveAs: {
+                    params.save_output_as_bam &&
+                    (
+                        params.save_mapped ||
+                        (params.skip_tools && params.skip_tools.split(',').contains('markduplicates')) ||
+                        !(params.tools && params.tools.split(',').contains('sentieon_dedup'))
+                    ) &&
+                    (meta.size * meta.num_lanes == 1) ? "mapped/${meta.id}/${it}" : null
+                }
+            ]
+        }
+
+
+        withName: "SENTIEON_BWAMEM" {
+            // Markduplicates Spark NEEDS name-sorted reads or runtime goes through the roof.
+            // However, currently SENTIEON_BWAMEM only supports coordinate sorting the reads.
+            ext.prefix       = { params.split_fastq > 1 ? "${meta.id}".concat('.').concat(reads.get(0).name.tokenize('.')[0]) : "${meta.id}.sorted" }
+            publishDir       = [
+                mode: params.publish_dir_mode,
+                path: { "${params.outdir}/preprocessing/" },
+                pattern: "*bam",
+                // Only save if save_output_as_bam AND
+                // (save_mapped OR no_markduplicates OR sentieon_dedup) AND
+                // only a single BAM file per sample
+                saveAs: {
+                    params.save_output_as_bam &&
+                    (
+                        params.save_mapped ||
+                        (params.skip_tools && params.skip_tools.split(',').contains('markduplicates')) ||
+                        !(params.tools && params.tools.split(',').contains('sentieon_dedup'))
+                    ) &&
+                    (meta.size * meta.num_lanes == 1) ? "mapped/${meta.id}/${it}" : null }
```

Comment:
I don't like having the ternary syntax on multiple lines like that, but for such a long condition, we might as well convert to a regular if

Reply:
How about this?

```
                saveAs: {
                    if (params.save_output_as_bam &&
                        (
                            params.save_mapped ||
                            (params.skip_tools && params.skip_tools.split(',').contains('markduplicates')) ||
                            !(params.tools && params.tools.split(',').contains('sentieon_dedup'))
                        ) && (meta.size * meta.num_lanes == 1)
                    ) { "mapped/${meta.id}/${it}" }
                    else { null }
                }
```

Reply:
is it working?